chainstate/src/detail/chainstateref/mod.rs

```diff
@@ -926,4 +913,30 @@ impl<'a, S: BlockchainStorageWrite, V: TransactionVerificationStrategy> Chainsta
         self.db_tx.add_block(block).map_err(BlockError::from).log_err()?;
         Ok(block_index)
     }
+
+    fn post_connect_tip(&mut self, tip_index: &BlockIndex) -> Result<(), BlockError> {
+        epoch_seal::update_epoch_seal(
+            &mut self.db_tx,
+            self.chain_config,
+            epoch_seal::BlockStateEvent::Connect(tip_index.block_height()),
+        )?;
+        epoch_seal::update_epoch_data(
+            &mut self.db_tx,
+            self.chain_config,
+            epoch_seal::BlockStateEventWithIndex::Connect(tip_index.block_height(), tip_index),
+        )
+    }
+
+    fn post_disconnect_tip(&mut self, tip_height: BlockHeight) -> Result<(), BlockError> {
+        epoch_seal::update_epoch_seal(
+            &mut self.db_tx,
+            self.chain_config,
+            epoch_seal::BlockStateEvent::Disconnect(tip_height),
+        )?;
+        epoch_seal::update_epoch_data(
+            &mut self.db_tx,
+            self.chain_config,
+            epoch_seal::BlockStateEventWithIndex::Disconnect(tip_height),
+        )
```

Comment:
Do these two function calls commute? My instinct would be fort hem to be applied in the opposite order compared to `connect_tip`.

Reply:
they don't, so the order doesn't matter